docs/DeveloperGuide.adoc

```diff
@@ -122,125 +119,647 @@ image::ModelClassDiagram.png[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+=== Module Management Feature
+The module feature manages the modules in Mod Manager and is represented by the `Module` class.
+A module has a `ModuleCode` and an optional `Description`.
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+It supports the following operations:
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+* add - Adds a module to Mod Manager.
+* list - Lists all modules in Mod Manager.
+* view - View information of a module in Mod Manager.
+* edit - Edits a module in Mod Manager.
+* delete - Deletes a module in Mod Manager.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+==== Implementation Details
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+===== Adding a module
+The add module feature allows users to add a module to Mod Manager.
+This feature is facilitated by `ModuleCommandParser`, `ModuleAddCommandParser` and `ModuleAddCommand`.
+The operation is exposed in the `Model` interface as `Model#addModule()`.
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+Given below is an example usage scenario and how the module add mechanism behaves at each step:
 
-image::UndoRedoState0.png[]
+1. The user executes the module add command and provides the module code and description of the module to be added.
+2. `ModuleAddCommandParser` creates a new `Module` based on the module code and description.
+3. `ModuleAddCommandParser` creates a new `ModuleAddCommand` based on the module.
+4. `LogicManager` executes the `ModuleAddCommand`.
+5. `ModManager` adds the module to the `UniqueModuleList`.
+6. `ModelManager` updates the `filteredModules` in `ModelManager`.
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+The following sequence diagram shows how the module add command works:
 
-image::UndoRedoState1.png[]
+.Sequence Diagram for `mod add` Command
+image::ModuleAddSequenceDiagram.png[]
```

Comment:
would be good if the arrows and constructors were bigger as they are rather difficult to see